### PR TITLE
Scheduler calendar ui #14

### PR DESCRIPTION
**Pull request by jassalss**
Hi Team! Scheduler Calendar is almost complete. More heavy testing left. It is working for small tests. Now It is connected to flask.

Some small tasks left like disabling previous dates. Please check my draft PR and share your views.

### CODE REVIEW

File: client/src/component/SchedularWidget.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                    return (
+                      <Box
+                        key={index}
+                        display="flex"
+                        flexDirection="row"
+                        mt={1}
+                      >
+                        <Box mr={1}>
+                          <Button variant="outlined" color="primary">
+                            {curr}
```
Comment (sina-jamshidi) [inline client/src/component/SchedularWidget.js]: Nice! This looks good!

File: client/src/pages/AvailabilitySettings.js [hunk 2]
```diff
@@ -1,15 +1,62 @@
...
+    let d = new Date(
+      now.getFullYear(),
+      now.getMonth(),
+      now.getDate(),
+      time.slice(0, 2),
+      time.slice(3),
+      0
+    );
+    let hours = d.getUTCHours();
+    let mintue = d.getUTCMinutes();
```
Comment (sina-jamshidi) [inline client/src/pages/AvailabilitySettings.js]: minute*

File: client/src/pages/AvailabilitySettings.js [hunk 3]
```diff
@@ -1,15 +1,62 @@
...
+    if (hours < 10) {
+      hours = "0" + hours;
+    }
+    if (mintue < 10) {
+      mintue = "0" + mintue;
+    }
+    return `${hours}:${mintue}`;
+  };
+  const handleFinish = (start, end, daysSelected) => {
+    const unique_url = localStorage.getItem("unique_url");
```
Comment (sina-jamshidi) [inline client/src/pages/AvailabilitySettings.js]: do we really need localstorage for this? these should be coming from the backend

File: client/src/pages/AvailabilitySettings.js [hunk 4]
```diff
@@ -1,15 +1,62 @@
...
+      time.slice(3),
+      0
+    );
+    let hours = d.getUTCHours();
+    let mintue = d.getUTCMinutes();
+    if (hours < 10) {
+      hours = "0" + hours;
+    }
+    if (mintue < 10) {
+      mintue = "0" + mintue;
```
Comment (sina-jamshidi) [inline client/src/pages/AvailabilitySettings.js]: minute*

File: client/src/pages/SchedulerCalendar.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from "react";
+import moment from "moment";
+import SchedulerWidget from "../component/SchedularWidget";
+const SchedulerCalendar = (props) => {
+  const [freeSlots, setFreeSlots] = useState([]);
+  const [dateSelected, setDateSelected] = useState(new Date());
+  const meetingTime = props.match.params.meetingTime;
+  useEffect(() => {
+    let access_token = localStorage.getItem("access_token");
+    let email = localStorage.getItem("email");
```
Comment (sina-jamshidi) [inline client/src/pages/SchedulerCalendar.js]: we should be using props, or if props have become too much of a hassle, a user Context or something. localstorage should really only be used for auth or very basic caching

**Review by sina-jamshidi — COMMENTED** [review]
Looks good in general!

File: client/src/pages/AvailabilitySettings.js (hunk 2 above)
Comment (jassalss) [reply client/src/pages/AvailabilitySettings.js]: fixed

File: client/src/pages/AvailabilitySettings.js (hunk 3 above)
Comment (jassalss) [reply client/src/pages/AvailabilitySettings.js]: yes. I am planning to use context to remove local storage issue.

File: client/src/pages/SchedulerCalendar.js (hunk 5 above)
Comment (jassalss) [reply client/src/pages/SchedulerCalendar.js]: yes. Context is coming to solve all issues related to local storage

File: client/src/contexts/CalendStore.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    propsForAuthWidget,
+    setPropsForAuthWidget,
+    email,
+    fromSignUp,
+    setFromSignUp,
+    uniqueUrl,
+    setUniqueUrl,
+    setUserId,
+    userId,
+  };
```
Comment (sina-jamshidi) [inline client/src/contexts/CalendStore.js:38]: looks good!

**Review by sina-jamshidi — APPROVED** [review]
Approved

**Pull request closed without merging**